ssl/ssl_ciph.c

```diff
@@ -1971,3 +1971,61 @@ int SSL_CIPHER_get_digest_nid(const SSL_CIPHER *c)
         return -1;
     return ssl_cipher_table_mac[i].nid;
 }
+
+int SSL_CIPHER_get_key_exchange(const SSL_CIPHER *c)
+{
+    if (c == NULL)
+        return 0;
+    if ((c->algorithm_mkey & SSL_kRSA))
```

Comment:
The double parens are on purpose. Some compilers will warn if a AND bitwise operation isn't in the form of ((A & B) != C), under the assumption that you really meant a logical AND. The double parentheses avoid that. I can always change it to be != 0 instead.